### PR TITLE
[#1936] Adapter Deletion: Show the Names of All Pipelines Using the Adapter and Allow One Click Deletion

### PR DESCRIPTION
**Pull request by muyangye**
### Purpose

See #1936 
### Remarks

PR introduces (a) breaking change(s): no

PR introduces (a) deprecation(s): no

### CODE REVIEW

**Comment by muyangye** [comment]
Hi @bossenti @dominikriemer, I noticed the cypress test failure. However, my local tests did not fail so I am not able to reproduce the problem:
82e6b6890ad6e30248d6fe4fa0764c4
I then run the failed fileStream and machineData smoke tests and logged the `delete-adapter` button to the console and clearly from the screenshot below that button doesn't exist anymore once the adapter is deleted:
aea1a958809d2d784d93ce70f830fce
312193e11a7ec6d500306571a9600e7
Can you test on your machines locally too? And if those 2 tests failed, can you please give me some information so that I can fix the problem? Thank you!

**Comment by dominikriemer** [comment]
Hi @muyangye great work! I think we just have some tests which currently randomly fail in CI. 
Your PR looks good :-)

File: streampipes-rest/src/main/java/org/apache/streampipes/rest/impl/connect/AdapterResource.java [hunk 1]
```diff
@@ -156,8 +162,27 @@ public Response deleteAdapter(@PathParam("id") String elementId) {
         LOG.error("Error while deleting adapter with id " + elementId, e);
         return ok(Notifications.error(e.getMessage()));
       }
+    } else if (!deleteAssociatedPipelines) {
+      List<String> namesOfPipelinesUsingAdapter = new ArrayList<String>();
+      for (String pipelineId : pipelinesUsingAdapter) {
+        namesOfPipelinesUsingAdapter.add(
+            StorageDispatcher.INSTANCE.getNoSqlStore().getPipelineStorageAPI().getPipeline(pipelineId).getName());
+      }
+      return Response.status(409).entity(String.join(", ", namesOfPipelinesUsingAdapter)).build();
```
[the remark this answers is not in the document]
Comment (muyangye) [reply streampipes-rest/src/main/java/org/apache/streampipes/rest/impl/connect/AdapterResource.java]: Just fixed the magic number. Thanks for pointing out!

**Review by tenthe — APPROVED** [review]
Hi @muyangye,

thank you very much for the PR. I tried it and it worked perfectly.

Cheers,
Philipp

**Comment by bossenti** [comment]
If merged the recent changes from dev, PR can be merged if CI passes

**Comment by muyangye** [comment]
> If merged the recent changes from dev, PR can be merged if CI passes

Hi @bossenti, I have tried multiple times but all cypress tests passed locally. The only way the 2 tests fail is when I set the timeout to a really low value such as 200. I see that the machine that runs CI is slower than my machine for every single test so I added more timeout. Can you trigger CI again? If it still doesn't pass then I sincerely don't understand what's wrong.

**Comment by muyangye** [comment]
If it's not the problem of delay, then I guess it's probably because of the base url in `cypress.config.ts` is not reflective of the ui change since it's port is 80 whereas the port of built ui is 8082. This might cause the backend delete adapter to expect a boolean parameter (though I don't think it's a problem since there's a default false) whereas if the ui is not updated clicking delete will call `deleteAdapter()` without passing any boolean which causes the problem.

As you can see here, it's really guesses after guesses since local cypress tests succeed but CI cypress tests fail. Is it possible to enable full logs for cypress tests in CI😳?

**Comment by bossenti** [comment]
Hey @muyangye, 

Sorry for not mentioning it earlier: we have some troubles with flaky e2e tests in the recent days. So your PR is very likely not the reason for the failing tests.
I really appreciate your endeavors fixing the issue and I apologize for these inconvenience. Can you please revert your recent change affecting the base url? 
I will then rerun the tests, if they still not pass we will take of this issue.

**Comment by dominikriemer** [comment]
Yes, the UI by default runs on port 80 and 8082 is only used for development. So if you change back the cypress base url to port 80, this shoudl be ok!

**Comment by muyangye** [comment]
> Hey @muyangye,
> 
> Sorry for not mentioning it earlier: we have some troubles with flaky e2e tests in the recent days. So your PR is very likely not the reason for the failing tests. I really appreciate your endeavors fixing the issue and I apologize for these inconvenience. Can you please revert your recent change affecting the base url? I will then rerun the tests, if they still not pass we will take of this issue.

I see, I thought it's my PR's problem because I observed other PRs passing CI and my PR failing the 2 tests that involved delete adapter. Thanks for notifying me of this. I have reverted both the port and timeout changes🙂.

**Comment by bossenti** [comment]
They sometimes pass and sometimes fail, let's cross fingers for this run 🤞
We need to fix the tests soon

**Comment by dominikriemer** [comment]
I ran the failing tests locally without any problem so we can merge the PR once the current CI rune is finished independent from its result.

**Pull request merged**